drivers/soundwire/bus_type.c

```diff
@@ -152,6 +154,9 @@ static int sdw_drv_remove(struct device *dev)
 	struct sdw_driver *drv = drv_to_sdw_driver(dev->driver);
 	int ret = 0;
 
+	slave->ops = NULL;
+	slave->probed = false;
+
 	if (drv->remove)
 		ret = drv->remove(slave);
```

Comment:
so in remove function slave->ops is not used ?   Why we set ops and probed to false before remove ?

Reply:
The slave->ops is only used at the bus level. If you don't first clear the value, it's possible that you have an interrupt while removing the driver and end-up calling a function in a driver that was removed.

It's quite possible we need a bus lock though to avoid race conditions on remove.